### ISSUE TITLE
getChildrenValues() for deep job hierarchies.

### ISSUE THREAD

**Issue by selvaganesh22** [taskforcesh/bullmq#682]
I am trying to implement the nested job structure ( for deep job hierarchies.) from https://docs.bullmq.io/guide/jobs/flows
getChildrenValues() of the parent job is returning only the immediate child data, missing all the other childrent data

**Comment by manast** [comment]
Can you provide a code snippet that shows what you mean?

**Comment by selvaganesh22** [comment]
const worker = new Worker(queueName, async job => {
            const childrenValues = await job.getChildrenValues();
            return childrenValues;
        });

**Comment by mltwist-brice** [comment]
@roggervalf Any progress on this one? Or do you need a better example? Basically, the issue is only the top-level parent can get the children values of all the sub jobs. But if I have a deeply nested hierarchy and the job two levels down might need the children values of the step just before it in order to know how to execute. Otherwise I have to chain together multiple flowProducers using steps to make a hacky controlled workflow.

**Comment by manast** [comment]
@mltwist-brice can't you just use job.getDependencies() to implement a recursive function that does what you want?

**Comment by mltwist-brice** [comment]
> @mltwist-brice can't you just use job.getDependencies() to implement a recursive function that does what you want?

@manast You are right. I didn't realize I got the job ids and their return values. I should be able to make that work. Thanks!
